pages/en/cookbook/grafting.mdx

```diff
@@ -0,0 +1,215 @@
+- The network should correspond to a indexed network being queried. Since we're running on Goerli testnet, the network is `goerli`
+- The `mapping` section defines the triggers of interest and the functions that should be run in response to those triggers. In this case, we are listening for the `Withdrawal` event and calling the `handleWithdrawal` function when it is emitted.
+
+## Grafting Manifest Definition
+
+Grafting requires adding two new items to the original subgraph manifest:
+
+```yaml
+---
+features:
+  - grafting # feature name
+graft:
+  base: Qm... # subgraph ID of base subgraph
+  block: 1502122 # block number
+```
+
+- `features:` is a list of all used [feature names](developing/creating-a-subgraph/#experimental-features).
+- `graft:` is a map of the `base` subgraph and the block to graft on to. The `block` is the block number to start indexing from. The Graph will copy the data of the base subgraph up to and including the given block and then continue indexing the new subgraph from that block on.
+
+The `base` and `block` values can be found by deploying two subgraphs: one for the base indexing and one with grafting
+
+## Deploying the Base Subgraph
+
+1. Go to [The Graph Studio UI](https://thegraph.com/studio/) and create a subgraph on Goerli testnet called `graft-example`
+2. Follow the directions in the `AUTH & DEPLOY` section on your subgraph page in the `graft-example` folder from the repo
+3. Once finished, verify the subgraph is indexing properly. If you run the following command in The Graph Playground
+
+```graphql
+{
+  withdrawals(first: 5) {
+    id
+    amount
+    when
+  }
+}
+```
+
+It returns something like this:
+
+```
+{
+  "data": {
+    "withdrawals": [
+      {
+        "id": "0x13098b538a61837e9f29b32fb40527bbbe63c9120c250242b02b69bb42c287e5-5",
+        "amount": "0",
+        "when": "1664367528"
+      },
+      {
+        "id": "0x800c92fcc0edbd26f74e19ad058c62008a47c7789f2064023b987028343dd498-3",
+        "amount": "0",
+        "when": "1664367648"
+      }
+    ]
+  }
+}
+```
+
+Once you have verified the subgraph is indexing properly, you can deploy the subgraph with grafting.
+
+## Deploying the Grafting Subgraph
+
+The graft replacement subgraph.yaml will have a new contract address
+
+1. Go to [The Graph Studio UI](https://thegraph.com/studio/) and create a subgraph on Goerli testnet called `graft-replacement`
+2. Create a new manifest. The `subgraph.yaml` for `graph-replacement` contains a different contract address and new information about how it should graft. These are the `block` of the [last event emitted](https://goerli.etherscan.io/tx/0x800c92fcc0edbd26f74e19ad058c62008a47c7789f2064023b987028343dd498) you care about by the old contract and the `base` of the old subgraph. The `base` subgraph ID is the `Deployment ID` of your original `graph-example` subgraph. You can find this in The Graph Studio UI.
```

Comment:
I'm also not sure how the events are handled. In our case, we have the same contract address (UUPS Proxy Pattern) but a different Event Signature. Would that require us to add a separate Event Handler to the base manifest?

Reply:
Could you clarify what you mean? Wouldn't the same event handler be fine since no code has changed, even though the addresses ([1](https://goerli.etherscan.io/address/0x4ed995e775d3629b0566d2279f058729ae6ea493),[2](https://goerli.etherscan.io/address/0xdb50720d920623a5bce4ffb65a06185dbf4d4c83)) are different?

Reply:
The contract code has changed, the new implementation (after upgrade) has different bytecode and different events.  IIUC a handler and an event have a 1:1 relationship. 
So the first version of the contract, the event had the following signature:
```
        - event: Retired(address,uint256)
```

But now it's:
```
        - event: Retired(address,uint256,uint256)
```
Would that not mean we need a different handler?

Reply:
The event signature `event Withdrawal(uint256 amount, uint256 when);` is the same for both, so do not think it would change

Reply:
I meant in our case, having two different signatures with the same name (due to a contract upgrade), is the main reason we are looking into Grafting. It would be great if you could cover that use case in the tutorial 🙏

Reply:
@marceljay in your case I don't think you need Grafting, you can just add a new event handler with the new signature

Reply:
It worked via Grafting. The old base subgraph uses a different handler implementation but maps to the same entity. This seems to work fine. I don't think it would work without it.

Reply:
@azf20 @t-proctor can you confirm the above? Meaning, that the code generate by `graph-cli` would overwrite the second event (with the same name), meaning it's not polymorphic?

Reply:
hey @marceljay it is hard to confirm what is happening with only the snippets, are you able to share the full codebase?
I think the cleanest approach here with the upgrade is probably to add a new dataSource with the new ABI (pointing at the same address) and the new handler, rather than merging into the existing dataSource

Reply:
This approach (two dataSources) wouldn't require grafting, right?

Reply:
yes no grafting required (this thread has gotten slightly off the original topic!)

Reply:
Thanks a lot, we can evaluate this internally. This would mean we need to have the legacy ABI in our build pipeline that is sharing artifacts within the mono-repo. I'd say there are pro's and con's.